src/mono/dlls/dbgshim/dbgshim.cpp

```diff
@@ -0,0 +1,294 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//*****************************************************************************
+// DbgShim.cpp
+//
+// This contains the APIs for creating a telesto managed-debugging session. These APIs serve to locate an
+// mscordbi.dll for a given telesto dll and then instantiate the ICorDebug object.
+//
+//*****************************************************************************
+
+#include <utilcode.h>
+
+#include "dbgshim.h"
+
+#include "palclr.h"
+
+#include <libloaderapi.h>
+
+/*
+
+// Here's a High-level overview of the API usage
+
+From the debugger:
+A debugger calls GetStartupNotificationEvent(pid of debuggee) to get an event, which is signalled when
+that process loads a Telesto.  The debugger thus waits on that event, and when it's signalled, it can call
+EnumerateCLRs / CloseCLREnumeration to get an array of Telestos in the target process (including the one
+that was just loaded). It can then call CreateVersionStringFromModule, CreateDebuggingInterfaceFromVersion
+to attach to any or all Telestos of interest.
+
+From the debuggee:
+When a new Telesto spins up, it checks for the startup event (created via GetStartupNotificationEvent), and
+if it exists, it will:
+- signal it
+- wait on the "Continue" event, thus giving a debugger a chance to attach to the telesto
+
+Notes:
+- There is no CreateProcess (Launch) case. All Launching is really an "Early-attach case".
```

Comment:
I don't think this is accurate for Mono.  But it is worthwhile to add a comment here to describe the flow of events for a Mono dbgshim and debuggee.